sdks/python/apache_beam/transforms/display/display_data.py

```diff
@@ -0,0 +1,133 @@
+"""
+DisplayData, its classes, interfaces and methods.
+"""
+
+from __future__ import absolute_import
+
+import calendar
+from datetime import datetime, timedelta
+import inspect
+import json
+
+__all__ = ['HasDisplayData', 'DisplayDataItem', 'DisplayData']
+
+class HasDisplayData(object):
+  """ Basic interface for elements that contain display data.
+
+  It contains only the display_data method.
+  """
+  def __init__(self, *args, **kwargs):
+    super(HasDisplayData, self).__init__(*args, **kwargs)
+
+  def display_data(self):
+    return {}
+
+  def _namespace(self):
+    return '{}.{}'.format(self.__module__, self.__class__.__name__)
+
+
+class DisplayData(object):
+  def __init__(self, namespace='__main__'):
+    self.namespace = namespace
+    self.items = []
+
+  def populate_items(self, display_data_dict):
+    for key, element in display_data_dict.items():
+      if isinstance(element, HasDisplayData):
+        subcomponent_display_data = DisplayData(element._namespace())
+        subcomponent_display_data.populate_items(element.display_data())
+        self.items += subcomponent_display_data.items
+        continue
+
+      if isinstance(element, dict):
+        self.items.append(
+            DisplayDataItem(self.namespace,
+                            key,
+                            DisplayDataItem._get_value_type(element['value']),
+                            element['value'],
+                            shortValue=element.get('shortValue'),
+                            url=element.get('url'),
+                            label=element.get('label')))
+        continue
+
+      # If it's not a HasDisplayData element,
+      # nor a dictionary, then it's a simple value
+      self.items.append(
+          DisplayDataItem(self.namespace,
+                          key,
+                          DisplayDataItem._get_value_type(element),
+                          element))
+
+  def output(self):
+    return [item.get_dict() for item in self.items]
+
+  @classmethod
```

Comment:
It's not odd to have a static method separate from the constructor that takes a different signature, but I do have to say it doesn't feel python to have a populate_items method distinct from the constructor. I would either (1, ideally) merge this into the constructor, if the items are knows at construction time or (2) name this add_item[s] if that's not possible.